### ISSUE TITLE
mapping.confirm doesn't escape correctly

### ISSUE THREAD

**Issue by figsoda** [hrsh7th/nvim-cmp#1213]
### FAQ

- [X] I have checked the [FAQ](https://github.com/hrsh7th/nvim-cmp/blob/main/doc/cmp.txt) and it didn't resolve my problem.

### Announcement

- [X] I have checked [Breaking change announcement](https://github.com/hrsh7th/nvim-cmp/issues/231).

### Minimal reproducible full config

```lua
local cmp = require("cmp")

vim.keymap.set("i", "<cr>", function() return "<cr>" end, { expr = true })

cmp.setup({
  mapping = {
    ["<cr>"] = cmp.mapping.confirm(), -- works without this line
  },
})
```

### Description

`mapping.confirm` doesn't escape correctly. Pressing enter types out `<cr>` instead of a new line

### Steps to reproduce

Open a file, press enter
```
a|b
```

### Expected behavior

```
a
b
```

### Actual behavior

```
a<cr>b
```

### Additional context

probably caused by breaking changes in nvim v0.8

also see https://github.com/neovim/neovim/pull/19598

**Comment by hrsh7th** [comment]
Thank you. I've fixed it.

**Comment by figsoda** [comment]
Thanks for the fix. Do you mind updating it on luarocks?

**Comment by hrsh7th** [comment]
I can't understand what updating luarocks. But feel free to do.

**Comment by briandipalma** [comment]
This seems to break nvim 0.7.

**Comment by appelgriebsch** [comment]
Yes, same here. As soon as an <CR> is pressed it gives an error trace....
```
E5108: Error executing lua: ...site/pack/packer/start/nvim-cmp/lua/cmp/utils/keymap.lua:245: invalid key: replace_keycodes
stack traceback:
        [C]: in function 'nvim_set_keymap'
        ...site/pack/packer/start/nvim-cmp/lua/cmp/utils/keymap.lua:245: in function 'set_map'
        ...site/pack/packer/start/nvim-cmp/lua/cmp/utils/keymap.lua:131: in function 'callback'
        ...m/site/pack/packer/start/nvim-cmp/lua/cmp/utils/misc.lua:13: in function 'fallback'
        ...te/pack/packer/start/nvim-cmp/lua/cmp/config/mapping.lua:181: in function 'on_keymap'
        ...re/nvim/site/pack/packer/start/nvim-cmp/lua/cmp/core.lua:145: in function 'callback'
        ...site/pack/packer/start/nvim-cmp/lua/cmp/utils/keymap.lua:117: in function <...site/pack/packer/start/nvim-cmp/lua/cmp/utils/keymap.lua:111>
```

**Comment by appelgriebsch** [comment]
Going back to commit `99ef854322d0de9269044ee197b6c9ca14911d96` solved the issue on Neovim 0.7

**Comment by LAST7** [comment]
Same problem here with Neovim0.7.0 build from source.
Updating the plugin doesn't seem to work, so I simply uninstall the old nvim and build nvim0.8.0 from source, and then the problem is gone.
